### PR TITLE
Allow `ExportPlugin` to add export messages; use in C# export plugin

### PR DESCRIPTION
**Pull request by raulsntos**
- Expose `EditorExportPlatform::add_message` as a method in `ExportPlugin` so plugins can add messages that will be displayed in the export log at the end of exporting.
- Use the new `ExportPlugin::add_message` in the C# export plugin to add error messages instead of a custom error dialog. Integrates the .NET error messages with the export log dialog so only one dialog is displayed.

add_message

### Alternative design

We could alternatively expose the method in `EditorExportPlatform` and add a `get_export_platform` method to `ExportPlugin` so we could retrieve the export platform from the plugin.

### CODE REVIEW

File: doc/classes/EditorExportPlatform.xml [hunk 1]
```diff
@@ -18,4 +18,18 @@
...
 	</methods>
+	<constants>
+		<constant name="EXPORT_MESSAGE_NONE" value="0" enum="ExportMessageType">
+			Invalid message type used as the default value when no type is specified.
+		</constant>
+		<constant name="EXPORT_MESSAGE_INFO" value="1" enum="ExportMessageType">
+			Message type for informational messages that have no effect on the export.
+		</constant>
+		<constant name="EXPORT_MESSAGE_WARNING" value="2" enum="ExportMessageType">
+			Message type for warning messages that should be addressed but still allow to complete the export successfully.
```
Comment (bruvzg) [inline doc/classes/EditorExportPlatform.xml:29]: Not sure about `complete the export successfully`, usually warnings are used when non-essential export step fails (changing icons, or code signing), some output is produced, but not necessarily fully usable.

**Review by bruvzg — APPROVED** [review]

File: doc/classes/EditorExportPlatform.xml (hunk 1 above)
Comment (raulsntos) [reply doc/classes/EditorExportPlatform.xml:29]: How about removing the word `successfully`?

```suggestion
			Message type for warning messages that should be addressed but still allow to complete the export.
```

**Comment by dsnopek** [comment]
> ### Alternative design
> We could alternatively expose the method in `EditorExportPlatform` and add a `get_export_platform` method to `ExportPlugin` so we could retrieve the export platform from the plugin.

I've been planning to eventually make a PR to add an `EditorExportPlatformExtension` class that would allow GDExtensions to add new `EditorExportPlatform`'s. (I started this in a branch a few months ago, but haven't found the time to finish it.)

This would be helpful in allowing folks to add support for exporting to console, for example, without needing to make a custom build of the editor.

In any case, as part of that work, we'd need to expose `EditorExportPlatform::add_message()` anyway. So, I personally quite like the alternative design, otherwise, we'll end up with two ways to do `add_message()`

**Comment by raulsntos** [comment]
That's a good point, the alternative design would also supersede https://github.com/godotengine/godot/pull/90556 since having access to the `EditorExportPlatform` would not require us to expose another way to get the OS name from `ExportPlugin`.

That said, as long as `EditorExportPlatform` is not accessible from `EditorPlugin` then even if both of these `add_message` methods are exposed, only one of them would be available depending on where you are.

**Comment by dsnopek** [comment]
> That said, as long as `EditorExportPlatform` is not accessible from `EditorPlugin` then even if both of these `add_message` methods are exposed, only one of them would be available depending on where you are.

Ah, yeah, that makes sense. Having `EditorPlugin::add_message()` available makes it more convenient to use when implementing an `EditorPlugin`, and there isn't really any harm in having it both places. :-)

**Comment by raulsntos** [comment]
- Superseded by https://github.com/godotengine/godot/pull/90782.

**Pull request closed without merging**